### PR TITLE
fix(tile): grow tiles along cross-axis 

### LINKED ISSUE #1359: [bug] rh-tile-group doesn't set the height correctly for multiple rh-tile(s)
### Describe the bug

Using `rh-tile-group` with multiple `rh-tile`s should auto adjust the height based on the content but currently we all the tiles have varied height.

### Which Component?

rh-tile

### Steps to reproduce

1. Go to rh-tile [demo ](https://ux.redhat.com/elements/tile/demo/) 
2. Inspect element and edit html
3. Update rh-tile's code with the snippet below
```html
<rh-tile-group style="
    display: flex;
    align-items: top;
    flex-wrap: wrap;
    column-gap: var(--rh-space-2xl, 32px);
    row-gap: var(--rh-space-2xl, 32px);
">
    <rh-tile style="width: 320px;">
      <img slot="image" src="https://fakeimg.pl/296x50" alt="296 X 50 placeholder image">
      <div slot="title">Title</div>
      <h2 slot="headline"><a href="#top">Link</a></h2>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      <div slot="footer">Suspendisse eu turpis elementum</div>
    </rh-tile>
    <rh-tile>
      <img slot="image" src="https://fakeimg.pl/296x50" alt="296 X 50 placeholder image">
      <div slot="title">Title</div>
      <h2 slot="headline"><a href="#top">Link</a></h2>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      <div slot="footer">Suspendisse eu turpis elementum</div>
    </rh-tile>  
</rh-tile-group>
```

4. Observe the variation in the height

### Expected behaviour

The height for each rh-tile inside rh-tile-group should be the same and based on the rh-tile with most content

### Screenshots

Screenshot 2023-11-22 at 8 28 53 AM

### Element HTML

_No response_

### Operating System (OS)

macOS

### Browser

Chrome, Firefox

### More browser information

_No response_

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by bennypowers**
Fixes #1359 

## What I did

1. make tile internal containers grow in the cross-axis, so that the host can flex as well. Make the host a flex container

## Testing Instructions

1. compare *all* the DP demos to production

### CODE REVIEW

**Comment by brianferry** [comment]
I see this also looks like it fixes the problem on the checkable demo where the tile takes up the full width instead of the 320px width it's set to on the page.  

Color context demo I am seeing the arrow now on the edge of the tile anymore

image

Seeing the same on compact demo.

**Comment by brianferry** [comment]
Outside of the arrow issue, everything looks good.  It looks like setting the rh-tile width to be calc(10 * --rh-space-2xl) also fixes the bugs demo but it is not included there originally.  Would that be enough to fix the issue in general or do we need this PR for that fix too?

**Review by marionnegp — CHANGES_REQUESTED** [review]
Under the Tile and Checkable demos, I'm able to select the disabled checkbox tile, which I don't think should be possible.

I'm seeing the same issue with the compact tile demos as @brianferry. 

Under the color context demo, the footer doesn't stick to the bottom in the basic card without an image.
Screenshot 2023-11-28 at 10 21 08 AM

Under the Full Width Images demo, the headline slot should be using `rh-color-interactive-blue-darker` and shouldn't be underlined in its default state.
Screenshot 2023-11-28 at 10 23 17 AM

Under the Tile Group Disabled and Tile Group Radio Disabled demos, should disabled tiles be focusable? I'm able to focus on the first tile only.

**Comment by bennypowers** [comment]
@marionnegp see #1302 for those issues

File: elements/rh-tile/demo/tile-group-flex.html [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<rh-tile-group>
+  <rh-tile>
+    <h2 slot="headline">
+      <a href="#top">Link</a>
+    </h2>
+    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
```
Comment (brianferry) [inline elements/rh-tile/demo/tile-group-flex.html:6]: Resolve to merge

**Review by brianferry — CHANGES_REQUESTED** [review]

**Review by brianferry — APPROVED** [review]
Code looks good, fixed the arrow issue I was seeing earlier.  👍

**Comment by bennypowers** [comment]
@marionnegp I addressed the layout issue with footer (full-height content)
and the issue of the headline link (that demo did not add the lightdom css, which I added back in just now)

The other issues should be fixed by #1302

**Review by marionnegp — COMMENTED** [review]
Thanks, @bennypowers! Lauded Gold Titleholder Mopes

**Review by marionnegp — APPROVED** [review]

**Pull request merged**